lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -134,15 +163,22 @@ newDBLayer fp = do
                                       Wallets
         -----------------------------------------------------------------------}
 
-        { createWallet = \(PrimaryKey wid) _cp meta ->
+        { createWallet = \(PrimaryKey wid) cp meta ->
             ExceptT $ runQuery conn $
-            handleConstraint (ErrWalletAlreadyExists wid) $
-            insert_ (mkWalletEntity wid meta)
+            handleConstraint (ErrWalletAlreadyExists wid) $ do
+                insert_ (mkWalletEntity wid meta)
+                insertCheckpoint wid cp
 
         , removeWallet = \(PrimaryKey wid) ->
             ExceptT $ runQuery conn $
             selectWallet wid >>= \case
-                Just _ -> Right <$> deleteCascadeWhere [WalTableId ==. wid]
+                Just _ -> Right <$> do
+                    -- fixme: deleteCascade is not working with persistent-sqlite.
+                    -- Therefore we need to delete related entities as well.
+                    deleteCheckpoints wid
```

Comment:
it would be great if we could implement cascade delete with TH on haskell side.

My understanding is that sqlite does support delete cascade but persistent-sqlite doesn't have this implemented?

Although, I agree that first iteration should be manual and we can implement this later TM.